components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/OAuthAppTenantResolverValve.java

```diff
@@ -0,0 +1,157 @@
+
+                if (StringUtils.isEmpty(clientId) && isOAuth10ARequest(request)) {
+                    clientId = request.getParameter(OAUTH_CONSUMER_KEY);
+                }
+
+                // If empty, try to get the client id from the authorization header.
+                if (StringUtils.isEmpty(clientId)) {
+                    if (OAuth2Util.isBasicAuthorizationHeaderExists(request)) {
+                        try {
+                            String[] credentials = OAuth2Util.extractCredentialsFromAuthzHeader(request);
+                            if (credentials != null) {
+                                clientId = credentials[0];
+                            }
+                        } catch (OAuthClientAuthnException e) {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Error while extracting credentials from authorization header.", e);
+                            }
+                        }
+                    }
+                }
+
+                if (StringUtils.isNotEmpty(clientId)) {
+                    try {
+                        OAuthAppDO oAuthAppDO = OAuth2Util.getAppInformationByClientIdOnly(clientId);
+                        appTenant = OAuth2Util.getTenantDomainOfOauthApp(oAuthAppDO);
+                    } catch (IdentityOAuth2Exception e) {
+                        log.error("Error while getting oauth app for client Id.", e);
+                    } catch (InvalidOAuthClientException e) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Error while getting oauth app for client Id: " + clientId, e);
+                        }
+                    }
+                }
+
+                // Set the tenant name to thread local properties.
+                if (StringUtils.isNotEmpty(appTenant)) {
+                    IdentityUtil.threadLocalProperties.get().put(TENANT_NAME_FROM_CONTEXT, appTenant);
+                }
+            }
+            getNext().invoke(request, response);
+        } finally {
+            // Clear thread local tenant name.
+            unsetThreadLocalContextTenantName();
+        }
+    }
+
+    /**
+     * Check whether the request is an OAuth request.
+     *
+     * @param request Http servlet request.
+     * @return True if the request is an OAuth request.
+     */
+    private boolean isOAuthRequest(Request request) {
+
+        initBaseUrls();
+        String requestUrl = request.getRequestURL().toString();
+        return StringUtils.isNotEmpty(requestUrl) && (requestUrl.startsWith(oAuth2ServerBaseURL) ||
+                requestUrl.startsWith(oAuthServerBaseURL));
+    }
+
+    /**
+     * Check whether the request is an OAuth 1.0 request.
+     *
+     * @param request Http servlet request.
+     * @return True if the request is an OAuth 1.0 request.
+     */
+    private boolean isOAuth10ARequest(Request request) {
```

Comment:
As Auth10 is deprecated shall we rename all to Auth20 and re - define the constants in Auth20?

Reply:
As this is not yet retired, still we need to retrieve and set the client ID for OAuth 1.0 use cases as well right?